### ISSUE TITLE
Deleting Files Via VimFiler (File Tree Navigation) on Windows 10

### ISSUE THREAD

**Issue by othmanalikhan** [wsdjeg/SpaceVim#1123]
## Summary
Deleting files through VimFiler on Windows does not work out of the box.
 
## Environment Information
- OS: Windows 10 Home, version 1703, OS Build 15063.786
- vim version: 8.0, 32-bit GUI version with OLE support

## The reproduce ways from Vim starting (Required!)
Launch GVim as usual, then navigate to any directory or file on VimFiler, then press "d" once to highlight the directory or file with a tick, and then press "d" again to delete that directory or file. The error message should pop up `[unite.vim] Please install rm.exe`.

## Solution
I've found a solution that works in my particular case for Windows. I'm leaving it here in case someone else comes across the same problem.

It seems that VimFiler uses some code from unite.vim to delete files and directories as implied by the error `[unite.vim] Please install rm.exe`. A quick (or not so quick) look at the [source code](https://github.com/Shougo/unite.vim/blob/master/autoload/unite/kinds/file_vimfiler_base.vim) (lines 11-16) indicates the cause of the problem (see below). 

```vim
" Global options definition. "{{{
call unite#util#set_default(
      \ 'g:unite_kind_file_delete_file_command',
      \ unite#util#is_windows() && !executable('rm') ? '' :
      \ executable('trash-put') ? 'trash-put $srcs' :
      \ executable('rmtrash') ? 'rmtrash $srcs' :
      \ 'rm $srcs')
```

Basically for Windows users, if the `rm` command does not exist in the shell that is linked to your Vim (default is CMD which does not have the `rm` command), then VimFiler cannot delete your files. The solution is to explicitly tell VimFiler what shell function to use when trying to delete a file.

In my case, I happen to be using Git Bash for Windows, so I have linked my Vim to use Git Bash instead of CMD, and so I use `rm` to delete files. Below is what I needed to add in my `.vimrc` file

```vim
let g:unite_kind_file_delete_directory_command="rmdir"  
let g:unite_kind_file_delete_file_command='rm'
set shell=C:\\Program\ Files\\Git\\bin\\bash.exe
```

**Comment by wsdjeg** [comment]
Thanks for your info, if user has installed the git-bash, then `executable('bash')` return 1? the path of git-bash's bin is added to PATH by default?

**Comment by othmanalikhan** [comment]
I've just checked it and `executable('bash')` returns 0 but `executable('git')` returns 1. Also, the path of Git Bash is not added to the system PATH but the path _specifically_ to the `git.exe` is added. So you would have to manually specify the path to `bash.exe` in `.vimrc`.

**Comment by wsdjeg** [comment]
hmm, This can only be done in user's custom config file, I do not know the path of `bash.exe` if it is not executable, BTW, not all user install the `bash.exe`.

I think window CMD provide command for delete file and directory, maybe this should be fixed in unite.

**Comment by othmanalikhan** [comment]
I just tried using CMD instead of Git Bash and it works perfectly fine with some extra lines in `.vimrc`. See below what I needed to include.
 
```
let g:unite_kind_file_delete_directory_command='del'
let g:unite_kind_file_delete_file_command='del'
```

Yes, should I therefore report this issue on the unite repository?

**Comment by wsdjeg** [comment]
yeah, :smile:

**Comment by othmanalikhan** [comment]
Very well. I shall close this issue then since it is complete. Thanks for the prompt replies 👍.
